ts/parts/DataLabels.ts

```diff
@@ -1521,18 +1521,26 @@ if (seriesTypes.pie) {
                     align: (labelPosition as any).alignment
                 };
 
+                // (#12985)
+                pointDataLabelsOptions =
+                    merge(options, point.options.dataLabels);
```

Comment:
Do we need to use `merge()` ? It's quite heavy.
In this case, simple `pick()` (because of numerical values for x/y) should be used.